processor/src/main/resources/org/mapstruct/ap/internal/model/assignment/AdderWrapper.ftl

```diff
@@ -8,9 +8,10 @@
 <#-- @ftlvariable name="" type="org.mapstruct.ap.internal.model.assignment.AdderWrapper" -->
 <#import "../macro/CommonMacros.ftl" as lib>
 <@lib.handleExceptions>
-  if ( ${sourceReference} != null ) {
-      for ( <@includeModel object=adderType.typeBound/> ${sourceLocalVarName} : ${sourceReference} ) {
+    <@lib.sourceLocalVarAssignment/>
+    <@lib.handleSourceReferenceNullCheck>
+        for ( <@includeModel object=adderType.typeBound/> ${sourceLoopVarName} : ${sourceReference} ) {
```

Comment:
Shouldn't this be:

```
for ( <@includeModel object=adderType.typeBound/> ${sourceLoopVarName} : ${sourceLocalVarName} ) {
```

Reply:
sigh.. pushed to quick :smile: .. thanks.. Will fix.